pytype/abstract/class_mixin.py

```diff
@@ -552,20 +568,31 @@ def _get_attrs_from_mro(self, cls_attrs, metadata_key):
 
     all_attrs = {}
     sub_attrs = []
+    type_params = {}
     attributes_to_ignore = set()
     for base_cls in reversed(self.mro[1:]):
       if not isinstance(base_cls, Class):
         continue
       # Some third-party dataclass implementations add implicit fields that
       # should not be considered inherited attributes.
       attributes_to_ignore.update(getattr(base_cls, "IMPLICIT_FIELDS", ()))
+      # Any subclass of a Parameterized dataclass must inherit attributes from
+      # its parent's init.
+      # See https://github.com/google/pytype/issues/1104
+      if isinstance(base_cls, _classes.ParameterizedClass):
+        type_params = base_cls._formal_type_parameters
+        base_cls = base_cls.base_cls
       if metadata_key in base_cls.metadata:
         sub_attrs.append([a for a in base_cls.metadata[metadata_key]
                           if a.name not in attributes_to_ignore])
     sub_attrs.append(cls_attrs)
     for attrs in sub_attrs:
       for a in attrs:
         all_attrs[a.name] = a
+
+    # type_params is only dict if it has been assigned in a dataclass context.
```

Comment:
why would type_params only be a dict for dataclasses?

Reply:
My previous change for comparing `type_param` as a dict is not needed anymore. I had issues last week when running tests, more particularly over `chex_overlay`, then `type_param` was not a dict. Tests are okay now. Reverted in lasted commit.